src/login/components/LoginFormLedger/LoginFormLedger.js

```diff
@@ -56,50 +66,63 @@ export default class LoginFormLedger extends React.Component {
   }
 
   renderActions = () => {
-    const disabled = this.props.disabled || !this.props.deviceInfo;
+    const disabled = (this.state.connected && this.state.open) ? this.props.disabled : true;
     const onClick = disabled ? null : this.handleLogin;
 
+    this.state.deviceError = (!disabled) ?
+      null :
+      this.state.deviceError;
+
     return (
       <div className={styles.actions}>
         <Button type="button" disabled={disabled} onClick={onClick}>Login</Button>
       </div>
     );
   }
 
-  renderDeviceInfo = () => {
-    const { deviceInfo } = this.props;
+  handleLogin = (event) => {
+    const { publicKey, onLogin } = this.props;
 
-    if (deviceInfo) {
-      return <p>Found USB {deviceInfo.manufacturer} {deviceInfo.product}</p>;
-    } else {
-      return <p>Searching for USB devices. Please plug in your Ledger to login.</p>;
-    }
+    event.preventDefault();
+    onLogin({ publicKey });
   }
 
-  renderDeviceStatus = () => {
-    const { publicKey } = this.props;
+  getDeviceInfo = () => {
+    const { deviceInfo, deviceError } = this.props;
```

Comment:
We're using props here, but then moving the props into the state?  I'm not following how this is solving the problem.